### PR TITLE
add reviewer functionality

### PR DESCRIPTION
**Pull request by alonisser**

### CODE REVIEW

File: studies/views/submitted_studies.py [hunk 1]
```diff
@@ -66,10 +66,17 @@ def get_queryset(self):
         qs = super().get_queryset()
         if self.action in ["my_studies"]:
             # for my studies we need to limit that
-            qs = qs.filter(submitter=self.request.user)
+            if hasattr(self.request.user, "profile") and self.request.user.profile.is_reviewer:
+                qs = qs.filter(Q(approval_status=ApprovalChoices.PENDING) | Q(submitter=self.request.user))
```
Comment (dani-efr) [inline studies/views/submitted_studies.py:70]: so if I have reviewer permissions I can see other submitters' pending studies as well as my own (which is the base case)?

File: studies/views/submitted_studies_experiments.py [hunk 2]
```diff
@@ -31,7 +31,9 @@ class SubmittedStudyExperiments(StudyRelatedPermissionsViewMixin, ModelViewSet,
     queryset = Experiment.objects.select_related("study", "study__approval_process", "study__submitter")
 
     def get_queryset(self):
-        qs = super().get_queryset().filter(study=self.kwargs.get("study_pk")).filter(study__submitter=self.request.user)
+        qs = super().get_queryset().filter(study=self.kwargs.get("study_pk"))
+        if not (hasattr(self.request.user, "profile") and self.request.user.profile.is_reviewer):
```
Comment (dani-efr) [inline studies/views/submitted_studies_experiments.py:35]: ואם כן?

File: studies/tests/test_study_submitting_endpoints.py [hunk 3]
```diff
@@ -163,6 +165,57 @@ def test_study_and_experiment_deletion_by_user(self):
...
+        study_id = study_res["id"]
+
+        experiments_res = self.get_experiments_for_study(study_id)
+        self.assertEqual(len(experiments_res), 0)
+        res_experiment = self.when_experiment_is_added_to_study_via_api(study_id=study_id)
+        experiment_id = res_experiment["id"]
+
+        self.given_user_authenticated("reviewer_user", "12345")
+
+        # verify this would fail with permission error
```
Comment (dani-efr) [inline studies/tests/test_study_submitting_endpoints.py:202]: so that reviewer can't edit, only view?

File: studies/permissions.py [hunk 4]
```diff
@@ -6,7 +6,8 @@ def has_object_permission(self, request, view, obj):
         # We're verifying that the user editing is only adding experiments, editing etc for studies he submitted
```
Comment (dani-efr) [inline studies/permissions.py:6]: they :)

File: studies/permissions.py [hunk 5]
```diff
@@ -6,7 +6,8 @@ def has_object_permission(self, request, view, obj):
...
         if not obj:
             return False
-        elif not obj.submitter == request.user:
+
+        elif (not obj.submitter == request.user) and (not request.user.profile.is_reviewer):
```
Comment (dani-efr) [inline studies/permissions.py:10]: how does this fit the test (with reviewer not being able to edit)?

**Review by dani-efr — COMMENTED** [review]
left some comments

**Review by dani-efr — DISMISSED** [review]

File: studies/views/submitted_studies.py (hunk 1 above)
Comment (alonisser) [reply studies/views/submitted_studies.py:70]: exactly!

File: studies/views/submitted_studies.py [hunk 6]
```diff
@@ -66,10 +66,17 @@ def get_queryset(self):
...
+            else:
+                qs = qs.filter(submitter=self.request.user)
         if self.action in ["update", "partial_update", "delete", "submit_to_review"]:
             # we can update only items still in pending status and that are 'mine"
-            qs = qs.filter(approval_status=ApprovalChoices.PENDING).filter(submitter=self.request.user)
+            pending_approval_qs = qs.filter(approval_status=ApprovalChoices.PENDING)
+
+            if not (hasattr(self.request.user, "profile") and self.request.user.profile.is_reviewer):
+                # reviewers can update all items
```
[the remark this answers is not in the document]
Comment (alonisser) [reply studies/views/submitted_studies.py:78]: true, I'll fix

File: studies/views/submitted_studies_experiments.py (hunk 2 above)
Comment (alonisser) [reply studies/views/submitted_studies_experiments.py:35]: then we don't need the extra filtering.. I've added a comment

**Pull request merged**